x-pack/plugins/index_management/public/application/components/component_templates/component_template_list/delete_modal.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { useComponentTemplatesContext } from '../component_templates_context';
+
+export const ComponentTemplatesDeleteModal = ({
+  componentTemplatesToDelete,
+  callback,
+}: {
+  componentTemplatesToDelete: string[];
+  callback: (data?: { hasDeletedComponentTemplates: boolean }) => void;
+}) => {
+  const { toasts, api } = useComponentTemplatesContext();
+  const numComponentTemplatesToDelete = componentTemplatesToDelete.length;
+
+  const handleDeleteComponentTemplates = () => {
+    api
+      .deleteComponentTemplates(componentTemplatesToDelete)
+      .then(({ data: { itemsDeleted, errors }, error }) => {
+        const hasDeletedComponentTemplates = itemsDeleted && itemsDeleted.length;
+
+        if (hasDeletedComponentTemplates) {
```

Comment:
nit: this is the typical example where I prefer to have an `i18nTexts` object declared on top of the component and use it in the JSX. It makes things much clearer to understand IMO

Reply:
👍 I'll keep this in mind. I'm going to leave as is for now, since I am not following this pattern elsewhere.